tests/test_routes/test_environment_route.py

```
@@ -0,0 +1,244 @@
+    with patch.object(
+        EnvController, "get_env_by_id", side_effect=HTTPException(status_code=404, detail="Environment not found")
+    ) as mock_read_env:
+        response = client.get("/environments/123")
+        assert response.status_code == 404
+        assert response.json() == {"detail": "Environment not found"}
+        mock_read_env.assert_called_once_with("123")
+
+def test_read_env_server_error():
+    with patch.object(
+        EnvController, "get_env_by_id", side_effect=Exception("error")
+    ) as mock_read_env:
```

Ruff

Local variable `mock_read_env` is assigned to but never used

(F841)